cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -1341,8 +1341,16 @@ queryResult eAcq = pure eAcq
 
 toEpochInfo :: EraHistory CardanoMode -> EpochInfo (Either Text)
 toEpochInfo (EraHistory _ interpreter) =
-  hoistEpochInfo (first (Text.pack . show ) . runExcept)
-    $ Consensus.interpreterToEpochInfo interpreter
+  hoistEpochInfo (first (Text.pack . show) . runExcept)
+    $ Consensus.interpreterToEpochInfo (Consensus.unsafeExtendSafeZone interpreter)
+
+newtype Unsafe a = Unsafe { unsafe :: a } deriving (Eq, Show)
```

Comment:
I like this use of a newtype. I think `Unsafe` is a bit too scary/general of a name though. Perhaps `AssumingNoUpcomingHardfork` (or something like that)?

Reply:
What about `newtype BeyondHorizonEpochInfo = BeyondHorizonEpochInfo (EpochInfo (Either Text))`